internal/controller/pkg/revision/reconciler.go

```diff
@@ -821,6 +853,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	// Reset status fields we care about that were clobbered by the Update
+	// above.
+	pr.SetStatusPackage(imagePath)
+	pr.SetAppliedImageConfigRefs(appliedImageConfigs...)
```

Comment:
This feels wrong, but I couldn't see another way to handle it. Hoping reviewers have a better solution :pray:

Reply:
:+1: I like that better than the previous solution as well; done.